### PR TITLE
openconnect: make host dependency more resilient

### PR DESCRIPTION
**Pull request by nmav**
Retry when resolveip fails as it seems to be causing issues on startup depending on various unpredictable parameters.

Resolves: 23185

Maintainer: me
Compile tested: N/A
Run tested: N/A

Description:
Retry when resolveip fails as it seems to be causing issues on startup depending on various unpredictable parameters.

### CODE REVIEW

**Comment by neheb** [comment]
I hink this needs a PKG_RELEASE bump.

File: net/openconnect/files/openconnect.sh [hunk 1]
```diff
@@ -76,11 +77,23 @@ proto_openconnect_setup() {
 	logger -t openconnect "initializing..."
 
 	[ -n "$interface" ] && {
+		local trials=10
+
 		logger -t "openconnect" "adding host dependency for $server at $config"
-		for ip in $(resolveip -t 10 "$server"); do
-			logger -t "openconnect" "adding host dependency for $ip at $config"
-			proto_add_host_dependency "$config" "$ip" "$interface"
+		resolveip -t "$server" >"$tmpfile"
```
Comment (efahl) [inline net/openconnect/files/openconnect.sh:83]: Checking sh status variable "$?" directly is sort of non-standard, I'd be inclined to write this something like this (which also gets rid of the double resolveip calls.  (Also, although it works in busybox ash, I'm not sure I've seen "let" used in any of the core sh scripts, I see everyone using the POSIX arithmetic.)
```
while resolveip -t "$server" >"$tmpfile" && [ "$trials" -gt 0 ]; do
        sleep 5
        trials=$((trials - 1))
done
```

Comment (gohai) [inline net/openconnect/files/openconnect.sh:83]: `-t` needs a timeout in seconds here to work

File: net/openconnect/files/openconnect.sh [hunk 2]
```diff
@@ -76,11 +77,23 @@ proto_openconnect_setup() {
...
+		while [ $? != 0 ] && [ $trials -gt 0 ]; do
+			sleep 5
+			let trials-=1
+			resolveip -t "$server" >"$tmpfile"
 		done
+
+		if [ -s "$tmpfile" ];then
```
Comment (gohai) [inline net/openconnect/files/openconnect.sh:90]: Style nitpick: elsewhere in the file there is a space before `then`

File: net/openconnect/files/openconnect.sh (hunk 1 above)
Comment (gohai) [reply net/openconnect/files/openconnect.sh:83]: Style nitpick: elsewhere in the file there is a space after `>`

File: net/openconnect/files/openconnect.sh [hunk 3]
```diff
@@ -76,11 +77,23 @@ proto_openconnect_setup() {
...
+
 		logger -t "openconnect" "adding host dependency for $server at $config"
-		for ip in $(resolveip -t 10 "$server"); do
-			logger -t "openconnect" "adding host dependency for $ip at $config"
-			proto_add_host_dependency "$config" "$ip" "$interface"
+		resolveip -t "$server" >"$tmpfile"
+		while [ $? != 0 ] && [ $trials -gt 0 ]; do
+			sleep 5
+			let trials-=1
+			resolveip -t "$server" >"$tmpfile"
```
[the remark this answers is not in the document]
Comment (gohai) [reply net/openconnect/files/openconnect.sh:87]: Style nitpick: elsewhere in the file there is a space after `>`

File: net/openconnect/files/openconnect.sh (hunk 1 above)
Comment (nmav) [reply net/openconnect/files/openconnect.sh:83]: Thanks

File: net/openconnect/files/openconnect.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (nmav) [reply net/openconnect/files/openconnect.sh]: Thanks

**Pull request merged**